Tests/BoutiqueTests/StoreTests.swift

```diff
@@ -8,7 +8,7 @@ final class StoreTests: XCTestCase {
     private var cancellables: Set<AnyCancellable> = []
 
     override func setUp() async throws {
-        store = Store<BoutiqueItem>(
+        store = try await Store<BoutiqueItem>(
```

Comment:
Would it be possible to make a separate `asyncStore` property for the tests that require an async Store? Ideally any tests where the behavior may differ significantly would have their own tests in case a change is made that introduces a bug only when the Store is synchronous, similar to how `StoredValue` and `AsyncStoredValue` have their own suite of tests.

Reply:
Hm, this one's tricky. The functionality in both async/non-async init cases would be the same, the only difference is that the later returns before `items` are retrieved. However since `setUp` also does a `removeAll()` it's not really relevant here.

I've [written a test](https://github.com/mergesort/Boutique/pull/39/commits/607e7ccb68f6d74305c3eda15349b8e713894309) that checks that creating a new store with `await` will have the `items` loaded. That test would fail when using a non-async init. However it's also tricky to test given that the Swift compiler prefers using the `async` version of the overload in an `async` context. In any case, I'm not sure if testing that behavior would be useful, since trying to fetch `items` that have not loaded is probably not desirable 😅

Reply:
What I was thinking was was to have two different Stores, one for async tests and one for sync tests. That way you wouldn't have to make assumptions about whether the store was sync or async, you just have two Stores and two sets of tests for a lot of the functionality.

I don't think it's necessary to rewrite _every test_, but I do think it would be good to have some baseline of tests that consider when the functionality could potentially differ, maybe such as one or two of the chained operation tests, `testInsertingItems()`, `testReadingItems()`, `testRemovingItems()`, and `testPublishedItemsSubscription()`.